ci-operator/step-registry/aws/provision/bastionhost/aws-provision-bastionhost-commands.sh

```diff
@@ -0,0 +1,448 @@
+
+acl authenticated proxy_auth REQUIRED
+acl CONNECT method CONNECT
+http_access allow authenticated
+http_port 3128
+EOF
+
+## PROXY Service
+cat > ${workdir}/squid-proxy.service << EOF
+[Unit]
+Description=OpenShift QE Squid Proxy Server
+After=network.target syslog.target
+
+[Service]
+Type=simple
+TimeoutStartSec=5m
+ExecStartPre=-/usr/bin/podman rm "squid-proxy"
+
+ExecStart=/usr/bin/podman run   --name "squid-proxy" \
+                                --net host \
+                                -p 3128:3128 \
+                                -p 3129:3129 \
+                                -v /srv/squid/etc:/etc/squid:Z \
+                                -v /srv/squid/cache:/var/spool/squid:Z \
+                                -v /srv/squid/log:/var/log/squid:Z \
+                                quay.io/crcont/squid
+
+ExecReload=-/usr/bin/podman stop "squid-proxy"
+ExecReload=-/usr/bin/podman rm "squid-proxy"
+ExecStop=-/usr/bin/podman stop "squid-proxy"
+Restart=always
+RestartSec=30
+
+[Install]
+WantedBy=multi-user.target
+EOF
+
+## ----------------------------------------------------------------
+# MIRROR REGISTORY
+# /opt/registry/auth/htpasswd
+# /opt/registry/certs/domain.crt
+# /opt/registry/certs/domain.key
+# /opt/registry/data
+# 
+## ----------------------------------------------------------------
+
+cat > ${workdir}/poc-registry.service << EOF
+[Unit]
+Description=OpenShift POC HTTP for PXE Config
+After=network.target syslog.target
+
+[Service]
+Type=simple
+TimeoutStartSec=5m
+ExecStartPre=-/usr/bin/podman rm "poc-registry"
+ExecStartPre=/usr/bin/chcon -Rt container_file_t /opt/registry
+
+ExecStart=/usr/bin/podman run   --name poc-registry -p 5000:5000 \
+                                --net host \
+                                -v /opt/registry/data:/var/lib/registry:z \
+                                -v /opt/registry/auth:/auth \
+                                -e "REGISTRY_AUTH=htpasswd" \
+                                -e "REGISTRY_AUTH_HTPASSWD_REALM=Registry Realm" \
+                                -e REGISTRY_AUTH_HTPASSWD_PATH=/auth/htpasswd \
+                                -v /opt/registry/certs:/certs:z \
+                                -e REGISTRY_HTTP_TLS_CERTIFICATE=/certs/domain.crt \
+                                -e REGISTRY_HTTP_TLS_KEY=/certs/domain.key \
+                                registry:2
+
+ExecReload=-/usr/bin/podman stop "poc-registry"
+ExecReload=-/usr/bin/podman rm "poc-registry"
+ExecStop=-/usr/bin/podman stop "poc-registry"
+Restart=always
+RestartSec=30
+
+[Install]
+WantedBy=multi-user.target
+EOF
+
+
+## ----------------------------------------------------------------
+# IGNITION
+## ----------------------------------------------------------------
+
+PROXY_PASSWORD="$(uuidgen | sha256sum | cut -b -32)"
+PROXY_CREDENTIAL="${BASTION_NAME}:$(openssl passwd -apr1 ${PROXY_PASSWORD})"
+PROXY_CREDENTIAL_CONTENT="$(echo -e ${PROXY_CREDENTIAL} | base64 -w0)"
```

Comment:
Can we do some similar things like vsphere, mount the credential from vault instead a random password?